### DISCUSSION TITLE
How to download different workflows to the specified directory

### DISCUSSION THREAD

**Discussion by xengine-qyt** [community/community#144510] [Actions]
### Select Topic Area

Question

### Body

I have three workflows, namely for Windows.yml, Linux.yml, and macOS.yml. Their corresponding platforms have x86 and arm. They are packaged and uploaded in the following way:
```
Name: Upload folder as artifact with ubuntu
Uses: actions/upload-artifact@v4
With:
Name: ProxyServiceApp-x64-Ubuntu
Path: Release/
```
and other 

Now I have newly created a release.yml, and I want to download the artifacts uploaded after compilation to a specified directory.

How can all the workflow files that were last uploaded and packaged be downloaded to the specified directory through actions/download-artifact@v4? And keep the original file names as well?

**Comment by versenilvis** [comment]
To download different workflows to a specified directory, the process depends on the type of workflows you are referring to. Assuming you're asking about downloading GitHub Actions workflows or similar automation files, here's how you can handle it:

1. Download GitHub Actions Workflows (from a GitHub repository)
If you want to download the workflow files (YAML files) from a GitHub repository to a specified directory on your local machine, follow these steps:

Using git:
Clone the repository: You can clone the entire repository containing the workflows using the git clone command. This will download all files in the repository, including the workflows.

bash
Copy code
git clone https://github.com/username/repository-name.git /path/to/your/directory
Replace:

https://github.com/username/repository-name.git with the URL of the repository.
/path/to/your/directory with the directory where you want to download the repository.
Navigate to the workflows folder: GitHub Actions workflows are usually stored in the .github/workflows directory of the repository. After cloning, you can navigate to this directory:

bash
Copy code
cd /path/to/your/directory/repository-name/.github/workflows
Using curl or wget (for downloading specific files):
If you only want to download the workflow files and not the entire repository, you can use curl or wget to download specific files.

Find the raw URL of the workflow file: Go to the GitHub repository, navigate to the workflow YAML file under .github/workflows/, and click on it. Then click on Raw to get the raw URL.

Download the file using curl or wget:

bash
Copy code
curl -o /path/to/your/directory/filename.yml https://raw.githubusercontent.com/username/repository-name/main/.github/workflows/filename.yml
Or, using wget:

bash
Copy code
wget -O /path/to/your/directory/filename.yml https://raw.githubusercontent.com/username/repository-name/main/.github/workflows/filename.yml
Replace:

https://raw.githubusercontent.com/username/repository-name/main/.github/workflows/filename.yml with the raw URL of the file.
/path/to/your/directory/filename.yml with the directory and filename where you want to save the file.

**Reply by xengine-qyt** [reply]
Sorry, I didn’t describe it clearly. I currently have three workflows, and I’ve already used actions/upload-artifact to upload the packages. Now, I have a new workflows.release.yml which is mainly used to publish all the packaged software to the GitHub release page. However, I’m not sure how to download all the files uploaded by actions/upload-artifact through actions/download-artifact, because I can only publish the release after downloading them. In other words, how can actions/download-artifact retrieve the files uploaded by other workflows?
